core/bin/zksync_api/src/bin/providers/dev_liquidity_provider.rs

```diff
@@ -0,0 +1,268 @@
+                        web::resource("/{contract_address}")
+                            .route(web::get().to(handle_get_coin_contract)),
+                    ),
+                ),
+            ),
+        );
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    #[test]
+    fn get_volume_for_whitelisted() {
+        let token = ("addr".to_string(), "name".to_string());
+        let storage = VolumeStorage::whitelisted_tokens(vec![token.clone()], 500.into());
+
+        let volume = storage.get_volume(&token.0);
+        assert_eq!(volume, 500.into());
+        let volume = storage.get_volume("wrong_addr");
+        assert_eq!(volume, 0.into())
+    }
+    #[test]
+    fn get_volume_for_blacklisted() {
+        let token = "addr".to_string();
+        let mut tokens = HashSet::new();
+        tokens.insert(token.clone());
+        let storage = VolumeStorage::blacklisted_tokens(tokens, 500.into());
+
+        let volume = storage.get_volume(&token);
+        assert_eq!(volume, 0.into());
+        let volume = storage.get_volume("another_token");
+        assert_eq!(volume, 500.into())
+    }
+}
+
+#[cfg(test)]
+mod handlers_tests {
+    use super::*;
+    use actix_web::{
+        body::{to_bytes, BoxAnyBody},
+        dev::AnyBody,
+        web,
+    };
+    use bigdecimal::FromPrimitive;
+
+    #[actix_web::test]
+    async fn get_asset_list() {
+        let expected_platform: AssetPlatform = AssetPlatform {
+            id: String::from(DEV_PLATFORM_ID),
+            chain_identifier: Some(ETHClientConfig::from_env().chain_id as i64),
+            name: String::from(DEV_PLATFORM_NAME),
+            shortname: String::from(DEV_PLATFORM_SHORTNAME),
+        };
+
+        let response = handle_get_asset_platforms().await.unwrap();
+        let status = response.status();
+
+        assert!(status.is_success());
+        let body = serde_json::from_slice::<Vec<AssetPlatform>>(
+            &to_bytes(AnyBody::Message(BoxAnyBody::from_body(
+                response.into_body(),
+            )))
+            .await
+            .unwrap(),
+        )
+        .unwrap(); // is this really the simplest way to get the body? seems so unnecessary to convert it to BoxAnyBody, etc!
+
+        let platform_option = body
+            .iter()
+            .find(|actual_platform| actual_platform.id.eq(&expected_platform.id));
+
+        assert!(platform_option.is_some());
+
+        let actual_platform = platform_option.unwrap();
+
+        assert_eq!(
+            actual_platform.chain_identifier,
+            expected_platform.chain_identifier
+        );
+        assert_eq!(actual_platform.name, expected_platform.name);
+        assert_eq!(actual_platform.shortname, expected_platform.shortname);
+    }
+
+    #[actix_web::test]
+    async fn get_coin_contract() {
+        let expected_volume = BigDecimal::from_f32(2.34567).unwrap();
+        let token_address = "0x2acc95758f8b5f583470ba265eb685a8f45fc9d5".to_string();
```

Comment:
Where are we getting this address?

Reply:
yeah but could we put a comment about it, I think that this is from mainnet inside the `/etc/tokens/`. @antomor this is the same from the other unit tests.

Reply:
I don't get the need. It's not like the tests will be ever ran against mainnet

Reply:
I gave it a random address and a comment